File: testview/sample.java
/**
 * 这是 Java 的多行文档注释 (Javadoc)
 * 展示了 Java 的各种字符串特性
 * 包括文本块（Java 15+）、多行注释等
 * 
 * @author Example
 * @version 1.0
 */
public class Sample {
  
  /**
   * 计算数字的平方
   * 
   * @param n 输入数字
   * @return 平方值
   */
  static int square(int n) {
    return n * n;
  }

  public static void main(String[] args) {
    /* 
     * 这是 Java 的传统多行注释
     * 可以跨越多行
     * 通常用于详细说明
     */
     
    // Java 15+ 文本块 (多行字符串)
    String multilineString = """
      这是 Java 文本块 (Text Block)
      可以包含多行内容
          保持缩进
      不需要转义 "引号" 或 '单引号'
      支持换行符
      """;
    
    // 包含变量插值的文本块 (通过字符串连接实现)
    String name = "Alice";
    int age = 30;
    String infoBlock = """
      用户信息:
         姓名: %s
         年龄: %d
      详细资料结束
      """.formatted(name, age);
    
    // 原始字符串概念的模拟 (Java没有真正的原始字符串)
    String rawLikeString = """
        C:\\Users\\Java\\Projects
        不需要转义反斜杠
        路径很清晰
        """;
        
    // 包含特殊字符的文本块
    String regexExample = """
      正则表达式示例:
        \\d{4}-\\d{2}-\\d{2}
      不需要过多转义
      """;
      
    // 在文本块中嵌入代码片段
    String codeSnippet = """
      public void example() {
          System.out.println("Hello");
      }
      """;
    
    System.out.println(multilineString);
    System.out.println(infoBlock);
    System.out.println(rawLikeString);
    System.out.println(regexExample);
    System.out.println(codeSnippet);
    
    String s = "Hello from Java";
    var values = List.of(2, 3, 4);
    int value = square(values.get(1)); // Function call
    System.out.println(s + " => " + value);
  }
}

import java.util.List;

File: testview/sample.py
"""
这是 Python 多行注释（文档字符串）
展示了 Python 的各种字符串特性
可以在这里详细说明代码功能
"""

def repeat(msg: str, n: int) -> str:
    """
    重复消息 n 次
    :param msg: 要重复的消息
    :param n: 重复次数
    :return: 重复后的字符串
    """
    return msg * n


# 多行字符串示例
multiline_string = """
这是第一行
这是第二行
还有第三行
可以包含任意换行
"""

# 原始字符串示例（常用于正则表达式）
raw_string = r'C:\Users\Python\Documents\no_escape_here'

# 另一种多行原始字符串
raw_multiline = r'''
原始多行字符串
不需要转义符号 \
甚至可以包含引号 " '
'''

# f-string 插值示例
name = "Alice"
age = 30
f_string_example = f"My name is {name} and I am {age} years old."

# 更复杂的 f-string 示例
calculation = f"Next year I will be {age + 1} years old."
coordinates = f"x={10}, y={20}"

# 使用三重引号嵌套引号
nested_quotes = """He said "Python is great!" and I agree."""

print(multiline_string)
print(raw_string)
print(raw_multiline)
print(f_string_example)
print(calculation)
print(coordinates)
print(nested_quotes)

text = " hello python "
cleaned = text.strip().title()  # String methods
print(repeat(cleaned + "!", 2))  # Function call